api/envoy/extensions/transport_sockets/proxy_protocol/v3/upstream_proxy_protocol.proto

```diff
@@ -17,6 +17,19 @@ option (udpa.annotations.file_status).package_version_status = ACTIVE;
 // [#protodoc-title: Upstream Proxy Protocol]
 // [#extension: envoy.transport_sockets.upstream_proxy_protocol]
 
+message CustomTlvMetadata {
+  repeated TlvEntry entries = 1;
+}
+
+message TlvEntry {
+  // type is the type of the TLV. The PP2 protocol expects this type to be a uint8_t.
+  uint32 type = 1 [(validate.rules).uint32 = {lt: 256}];
+  // value is the value of the TLV.
+  //
+  // TODO(tim): should this be a bytes field instead?
+  string value = 2 [(validate.rules).string = {min_len: 1}];
```

Comment:
Still an open question after re-reading the protocol documentation:

```
The information in those bytes will be arranged in Type-Length-Value (TLV
vectors) in the following format.  The first byte is the Type of the vector.
The second two bytes represent the length in bytes of the value (not included
the Type and Length bytes), and following the length field is the number of
bytes specified by the length.

        struct pp2_tlv {
            uint8_t type;
            uint8_t length_hi;
            uint8_t length_lo;
            uint8_t value[0];
        };
```

I think changing the type from string to bytes would push some slight complexity to consumers of this metadata, which doesn't seem unreasonable.

Reply:
I think it makes sense to change the value to bytes.